coral/src/app/features/components/table-filters/OperationTypeFilter.tsx

```diff
@@ -0,0 +1,58 @@
+import { NativeSelect } from "@aivenio/aquarium";
+import { useSearchParams } from "react-router-dom";
+import {
+  requestOperationTypeNameMap,
+  operationTypeList,
+} from "src/app/features/approvals/utils/request-operation-type-helper";
+import { RequestOperationType } from "src/domain/requests/requests-types";
+import { ResolveIntersectionTypes } from "types/utils";
+
+type RequestOperationTypeOptions = ResolveIntersectionTypes<
+  RequestOperationType | "ALL"
+>;
+
+function OperationTypeFilter() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const operationType =
+    (searchParams.get("operationType") as RequestOperationTypeOptions | null) ??
+    "ALL";
+
+  const handleChangeOperationType = (
+    nextOperationType: RequestOperationTypeOptions
+  ) => {
+    if (nextOperationType === "ALL") {
+      searchParams.delete("operationType");
+      searchParams.set("page", "1");
+      setSearchParams(searchParams);
```

Comment:
You could move these after these lines after the if statement, so they are not repeated in each block.